spaces_keys.go

```diff
@@ -0,0 +1,164 @@
+package godo
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const spacesKeysBasePath = "v2/spaces/keys"
+
+// SpacesKeysService is an interface for managing Spaces keys with the DigitalOcean API.
+type SpacesKeysService interface {
+	ListSpacesKeys(context.Context, *ListOptions) ([]*SpacesKey, *Response, error)
+	UpdateSpacesKey(context.Context, string, *SpacesKeyUpdateRequest) (*SpacesKey, *Response, error)
+	CreateSpacesKey(context.Context, *SpacesKeyCreateRequest) (*SpacesKey, *Response, error)
+	DeleteSpacesKey(context.Context, string) (*Response, error)
+}
+
+// SpacesKeysServiceOp handles communication with the Spaces key related methods of the
+// DigitalOcean API.
+type SpacesKeysServiceOp struct {
+	client *Client
+}
+
+var _ SpacesKeysService = &SpacesKeysServiceOp{}
+
+// Permission represents a permission for a Spaces grant
+type Permission string
+
+const (
+	// PermissionRead grants read-only access to the Spaces bucket
+	PermissionRead Permission = "read"
+	// PermissionReadWrite grants read and write access to the Spaces bucket
+	PermissionReadWrite Permission = "readwrite"
+	// PermissionFullAccess grants full access to the Spaces bucket
+	PermissionFullAccess Permission = "fullaccess"
+)
+
+// Grant represents a Grant for a Spaces key
+type Grant struct {
+	Bucket     string     `json:"bucket"`
+	Permission Permission `json:"permission"`
+}
+
+// SpacesKey represents a DigitalOcean Spaces key
+type SpacesKey struct {
+	Name      string `json:"name"`
+	AccessKey string `json:"access_key"`
+	SecretKey string `json:"secret_key"`
+	CreatedAt string `json:"created_at"`
+}
+
+// SpacesKeyRoot represents a response from the DigitalOcean API
+type spacesKeyRoot struct {
+	Key *SpacesKey `json:"key"`
+}
+
+// SpacesKeyCreateRequest represents a request to create a Spaces key.
+type SpacesKeyCreateRequest struct {
+	Name   string   `json:"name"`
+	Grants []*Grant `json:"grants"`
+}
+
+// SpacesKeyUpdateRequest represents a request to update a Spaces key.
+type SpacesKeyUpdateRequest struct {
+	Name   string   `json:"name"`
+	Grants []*Grant `json:"grants"`
+}
+
+// spacesListKeysRoot represents a response from the DigitalOcean API
+type spacesListKeysRoot struct {
+	Keys  []*SpacesKey `json:"keys,omitempty"`
+	Links *Links       `json:"links,omitempty"`
+	Meta  *Meta        `json:"meta"`
+}
+
+// CreateSpacesKey implements SpacesKeysService.
+func (s *SpacesKeysServiceOp) CreateSpacesKey(ctx context.Context, createRequest *SpacesKeyCreateRequest) (*SpacesKey, *Response, error) {
```

Comment:
```suggestion
func (s *SpacesKeysServiceOp) Create(ctx context.Context, createRequest *SpacesKeyCreateRequest) (*SpacesKey, *Response, error) {
```

We can remove the `SpacesKey` suffix as it gets a bit repetitive, e.g. compare ` client.SpacesKeys.CreateSpacesKey` to `client.SpacesKeys.Create`